server/play-server/src/main/scala/sttp/tapir/server/play/PlayRequestBody.scala

```diff
@@ -117,3 +126,5 @@ private[play] class PlayRequestBody(serverOptions: PlayServerOptions)(implicit
 
   private def playRequest(serverRequest: ServerRequest) = serverRequest.underlying.asInstanceOf[Request[Source[ByteString, Any]]]
 }
+
+class PlayBodyParserException(val result: Result) extends Exception
```

Comment:
One thing I'm contemplating is that this mechanism escapes our response formatting. Normally, there's a way to change the format of all your error responses using `CustomiseInterceptors.defaultHandlers` so that errors can always be returned as json, for example.

Maybe it would make sense to introduce a top-level exception in `serverCore` with special handling in `ExceptionInterceptor` (sth like `CustomErrorException` but the name is a bit weird :) ), which would overwrite the default status code & error message? Maybe other interpreters can benefit from this as well.

What do you think?

Reply:
> One thing I'm contemplating is that this mechanism escapes our response formatting

You can have a Play filter configured, that can immediately return a response before it reaches Tapir. I was thinking of these bodyParser errors in the same way... 

> so that errors can always be returned as json, for example.

you can still achieve that using the Play `HttpErrorHandler`... I think is a matter of deciding/documenting that some errors will not reach Tapir.

> Maybe it would make sense to introduce a top-level exception in serverCore

What would this exception contain? Because in this case the only thing I have is a Play `Result`.

Reply:
The exception would have to contain the status code & the string message.

I guess it's a question whether we leave the error handling to play or to tapir. Tapir handles quite a lot of errors by itself - starting with arbitrary exceptions (via the `ExceptionInterceptor`, unless it's explicitly disabled), ending with decode failures. What I'd like to avoid is having two places where you need to configure this. So in this solution our interceptor would replace Play's `HttpErrorHandler`

Reply:
> The exception would have to contain the status code & the string message.

What if the play `Result` contains a streaming response (i.e the `result.body` is not `HttpEntity.Strict`)? 

> I guess it's a question whether we leave the error handling to play or to tapir.

Why not both? That's why I mentioned the Play filters. There are some errors that will be handled by play infrastructure and some others that will be handled by tapir infrastructure...

Reply:
Ok, let's keep this solution for now:)